backbone.js

```diff
@@ -1102,20 +1102,14 @@
         var index = this.indexOf(model);
         this.models.splice(index, 1);
         this.length--;
-
-        // Remove references before triggering 'remove' event to prevent an
-        // infinite loop. #3693
-        delete this._byId[model.cid];
-        var id = this.modelId(model.attributes);
-        if (id != null) delete this._byId[id];
+        this._removeReference(model, options);
+        removed.push(model);
 
         if (!options.silent) {
           options.index = index;
           model.trigger('remove', model, this, options);
+          this._onModelEvent('remove', model, this, options);
```

Comment:
:eyeroll:

The ones that are meant to be overridden are usually obvious, or their
utility becomes clear when you need them. They don't get in your way if you
don't need them.

I can't think of a reason why you'd override _onModelEvent. It seems to be
there mostly to make it easier to unbind later.

Theres no issue using _onModelEvent but I like the symmetry of the two
triggers in a row.

On Mon, Sep 28, 2015, 10:34 Justin Ridgewell <notifications@github.com>
wrote:

> In backbone.js
> <https://github.com/jashkenas/backbone/pull/3806#discussion_r40558531>:
>
> >
> >          if (!options.silent) {
> >            options.index = index;
> >            model.trigger('remove', model, this, options);
> > +          this._onModelEvent('remove', model, this, options);
>
> Since it could be overridden. We're not exactly clear on what is
> overridable and what's not. #3703
> <https://github.com/jashkenas/backbone/pull/3703>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jashkenas/backbone/pull/3806/files#r40558531>.
>
-- 

Adam K (mobile)